Source/JavaScriptCore/wasm/WasmInstance.cpp

```diff
@@ -313,27 +330,51 @@ void Instance::copyElementSegment(const Element& segment, uint32_t srcOffset, ui
 
     for (uint32_t srcIndex = srcOffset; srcIndex < length; ++srcIndex) {
         const auto dstIndex = srcIndex - srcOffset;
+        const auto initType = segment.initTypes[srcIndex];
+        const auto initialBitsOrIndex = segment.initialBitsOrIndices[srcIndex];
 
         // Represent the null function as the null JS value
-        if (Element::isNullFuncIndex(segment.functionIndices[srcIndex])) {
+        if (initType == Element::InitializationType::FromRefNull) {
             values[dstIndex] = static_cast<uint64_t>(JSValue::encode(jsNull()));
             continue;
         }
 
-        // FIXME
-        // This will have to be updated to handle element types other than function references
-        // when https://bugs.webkit.org/show_bug.cgi?id=251874 is fixed
-        uint32_t functionIndex = segment.functionIndices[srcIndex];
-
-        // A wrapper for this function should have been created during parsing.
-        // A future optimization would be for the parser to not create the wrappers,
-        // and create them here dynamically instead.
-        JSValue value = getFunctionWrapper(functionIndex);
-        ASSERT(value.isCallable());
-        values[dstIndex] = static_cast<uint64_t>(JSValue::encode(value));
+        if (initType == Element::InitializationType::FromRefFunc) {
+            uint32_t functionIndex = static_cast<uint32_t>(initialBitsOrIndex);
+
+            // A wrapper for this function should have been created during parsing.
+            // A future optimization would be for the parser to not create the wrappers,
+            // and create them here dynamically instead.
+            JSValue value = getFunctionWrapper(functionIndex);
+            ASSERT(value.isCallable());
+            values[dstIndex] = static_cast<uint64_t>(JSValue::encode(value));
+            continue;
+        }
+
+        if (initType == Element::InitializationType::FromGlobal) {
+            values[dstIndex] = loadI64Global(initialBitsOrIndex);
+            continue;
+        }
+
+        ASSERT(initType == Element::InitializationType::FromExtendedExpression);
+        uint64_t result;
```

Comment:
Can this be fixed?

Reply:
Oh whoops, thanks for catching this. Currently it's not expected that this should ever fail, so I'll put in a `RELEASE_ASSERT` but once I fix https://bugs.webkit.org/show_bug.cgi?id=264454 it will change to throw an exception in rare cases (ditto for the other call of `evaluateConstantExpression`).